src/messaging/ReliableMessageManager.h

```diff
@@ -99,27 +100,33 @@ class ReliableMessageManager
     int TestGetCountRetransTable();
     void TestSetIntervalShift(uint16_t value) { mTimerIntervalShift = value; }
 
-public:
-    // public functions for ReliableMessageProtocol internal usage
-    CHIP_ERROR SendMessage(ReliableMessageContext * context, System::PacketBuffer * msgBuf, uint16_t sendFlags);
-    CHIP_ERROR SendMessage(ReliableMessageContext * context, uint32_t profileId, uint8_t msgType, System::PacketBuffer * msgBuf,
-                           BitFlags<uint16_t, SendMessageFlags> sendFlags);
-
 private:
     chip::System::Layer * mSystemLayer;
+    ExchangeContext * mContextPool;
+    SecureSessionMgr * mSessionMgr;
     uint64_t mTimeStampBase;                  // ReliableMessageProtocol timer base value to add offsets to evaluate timeouts
     System::Timer::Epoch mCurrentTimerExpiry; // Tracks when the ReliableMessageProtocol timer will next expire
     uint16_t mTimerIntervalShift;             // ReliableMessageProtocol Timer tick period shift
 
     /* Placeholder function to run a function for all exchanges */
     template <typename Function>
     void ExecuteForAllContext(Function function)
-    {}
+    {
+        ExchangeContext * ec = mContextPool;
+
+        if (ec != nullptr)
```

Comment:
If this class took a reference in the constructor, we would never have null here.